kafka-connect-s3/src/main/java/io/confluent/connect/s3/storage/S3OutputStream.java

```diff
@@ -168,6 +168,9 @@ public void commit() throws IOException {
       if (multiPartUpload == null && buffer.hasRemaining()) {
         ObjectMetadata metadata = newObjectMetadata();
         metadata.setContentLength(buffer.position());
+        if (CompressionType.GZIP == compressionType) {
```

Comment:
Yes, I noticed that inside S3 the file type is `application/octet-stream` which is the default one when it should be `application/gzip`. Also, we might also need to add MD5 here for verification during uploading, wdyt?

Reply:
Do you mean "manually" calculate md5 and assign the tag to key?

Reply:
I have read the javadoc and it seems that this method specifically calculates the MD5 for us. This is also the reason why we must specify the `setContentLength` and `setContentType` because this can't be inferred from the stream